test/passing/js_source.ml.ref

```diff
@@ -103,7 +103,7 @@ let () =
       done ) [@foo]];
   [%foo (assert true [@foo])];
   [%foo (lazy x) [@foo]];
-  [%foo (object end) [@foo]];
+  [%foo (object end [@foo])];
```

Comment:
Question: Am I correct that the output before still parses correctly, or is the sanity checking not tight enough?